src/app/tabs/view-foreign-profile/view-foreign-profile.page.ts

```diff
@@ -47,6 +62,14 @@ export class ViewForeignProfilePage implements OnInit {
       this.tagValidations = result.map((validation) => validation.comment);
     });
   }
+  addContact() {
+    let persistence = new Persistence();
```

Comment:
Would change `let` to `const`, but because we want to rewrite the whole persistence stuff anyway, this isn't necessary.